joint_state_publisher/joint_state_publisher/joint_state_publisher

```diff
@@ -214,9 +214,14 @@ class JointStatePublisher():
                 elif name in self.dependent_joints:
                     param = self.dependent_joints[name]
                     parent = param['parent']
-                    joint = self.free_joints[parent]
                     factor = param.get('factor', 1)
                     offset = param.get('offset', 0)
+                    while parent in self.dependent_joints:
+                        param = self.dependent_joints[parent]
+                        parent = param['parent']
+                        factor *= param.get('factor', 1)
+                        # the offset is relative only to the first parent
```

Comment:
Hi @alextoind, thank you for your answer, and sorry for my late reply

> I thought about it for a while and I concluded that a user is expecting to set the speed of the joint w.r.t. the whole chain of recursive joints, but the initial placement is logical to be w.r.t. the immediate first parent from which the URDF is computing the distance.

I would suggest to implement exactly what the spec/doc states instead of guessing what users have in mind (they might all have a different thing in mind).


[Here is the mimic doc](http://wiki.ros.org/urdf/XML/joint):
this tag is used to specify that the defined joint mimics another existing joint. The value of this joint can be computed as value = multiplier * other_joint_value + offset.

> Example: imagine to have three joints Ja mimics Jb which mimics Jc. And suppose that Jb has an offset of pi/2 w.r.t. Jc. If you sum up all the offset, even if Ja has no offset w.r.t. Jb, it ends to inherit the other offsets and to refer them to Jb.

I'm not sure to understand. Is this the example you imagined?
```
 <joint name="Jc" type="continuous">
 </joint>
 <joint name="Jb" type="continuous">
     <mimic joint="Jc" multiplier="x" offset="y"/>
 </joint>
 <joint name="Ja" type="continuous">
    <mimic joint="Jb" multiplier="x'" offset="y'"/>
 </joint>
````

Then I would expect the following behavior:
```
Jb_value == x * Jc_value + y
Ja_value == x' * Jb_value + y' == (x * x') * Jc_value + (x' * y + y')
```
Do you agree?